x/concentrated-liquidity/incentives_test.go

```diff
@@ -1832,3 +1876,874 @@ func (s *KeeperTestSuite) TestInitOrUpdatePositionUptime() {
+			// uptimes the position qualifies for.
+			expectedIncentivesClaimed: expectedIncentivesFromUptimeGrowth(uptimeHelper.hundredTokensMultiDenom, DefaultLiquidityAmt, oneDayFreeze, defaultMultiplier),
+		},
+		"(curr < lower < upper) no uptime growth inside or outside range, 1W freeze duration": {
+			currentTick: 0,
+			positionParams: positionParameters{
+				owner:          ownerWithValidPosition,
+				lowerTick:      1,
+				upperTick:      2,
+				liquidity:      DefaultLiquidityAmt,
+				joinTime:       defaultJoinTime,
+				freezeDuration: oneWeekFreeze,
+				collectTime:    defaultJoinTime.Add(100),
+			},
+			numPositions: 1,
+
+			expectedIncentivesClaimed: sdk.Coins(nil),
+		},
+		"(curr < lower < upper) uptime growth outside range but not inside, 1W freeze duration": {
+			currentTick:              0,
+			addedUptimeGrowthOutside: uptimeHelper.hundredTokensMultiDenom,
+			positionParams: positionParameters{
+				owner:          ownerWithValidPosition,
+				lowerTick:      1,
+				upperTick:      2,
+				liquidity:      DefaultLiquidityAmt,
+				joinTime:       defaultJoinTime,
+				freezeDuration: oneWeekFreeze,
+				collectTime:    defaultJoinTime.Add(100),
+			},
+			numPositions: 1,
+
+			// Since there was no growth inside the range, we expect no incentives to be claimed
+			expectedIncentivesClaimed: sdk.Coins(nil),
+		},
+		"(curr < lower < upper) uptime growth inside range but not outside, 1W freeze duration": {
+			currentTick:             0,
+			addedUptimeGrowthInside: uptimeHelper.hundredTokensMultiDenom,
+			positionParams: positionParameters{
+				owner:          ownerWithValidPosition,
+				lowerTick:      1,
+				upperTick:      2,
+				liquidity:      DefaultLiquidityAmt,
+				joinTime:       defaultJoinTime,
+				freezeDuration: oneWeekFreeze,
+				collectTime:    defaultJoinTime.Add(100),
+			},
+			numPositions: 1,
+
+			// Since there is no other existing liquidity, we expect all of the growth inside to accrue to be claimed for the
+			// uptimes the position qualifies for.
+			expectedIncentivesClaimed: expectedIncentivesFromUptimeGrowth(uptimeHelper.hundredTokensMultiDenom, DefaultLiquidityAmt, oneWeekFreeze, defaultMultiplier),
+		},
+		"(curr < lower < upper) uptime growth both inside and outside range, 1W freeze duration": {
+			currentTick:              0,
+			addedUptimeGrowthInside:  uptimeHelper.hundredTokensMultiDenom,
+			addedUptimeGrowthOutside: uptimeHelper.hundredTokensMultiDenom,
+			positionParams: positionParameters{
+				owner:          ownerWithValidPosition,
+				lowerTick:      1,
+				upperTick:      2,
+				liquidity:      DefaultLiquidityAmt,
+				joinTime:       defaultJoinTime,
+				freezeDuration: oneWeekFreeze,
+				collectTime:    defaultJoinTime.Add(100),
+			},
+			numPositions: 1,
+
+			// Since there is no other existing liquidity, we expect all of the growth inside to accrue to be claimed for the
+			// uptimes the position qualifies for.
```

Comment:
```suggestion
			// Since there is no other existing liquidity, we expect all of the growth inside to accrue to be claimed for the
			// uptimes the position qualifies for. At the same time, growth outside does not affect the current position's incentive rewards.
```